sdk/go/arvadosclient/arvadosclient.go

```diff
@@ -20,13 +20,31 @@ import (
 var MissingArvadosApiHost = errors.New("Missing required environment variable ARVADOS_API_HOST")
 var MissingArvadosApiToken = errors.New("Missing required environment variable ARVADOS_API_TOKEN")
 
-type ArvadosApiError struct {
-	error
-	HttpStatusCode int
-	HttpStatus string
+// Indicates an error that was returned by the API server.
+type RemoteApiServerError struct {
+	// Address of server returning error, of the form "host:port".
+	ServerAddress string
+
+	// Components of server response.
+	HttpStatusCode    int
+	HttpStatusMessage string
+
+	// Additional error details from response body.
+	ErrorDetails []string
 }
 
-func (e ArvadosApiError) Error() string { return e.error.Error() }
+func (e RemoteApiServerError) Error() string {
+	s := fmt.Sprintf("API server (%s) returned %d: %s.",
```

Comment:
Thanks! I switched to the following two flavors:

With details:

    arvados API server error: #<NoMemoryError: failed to allocate memory> (422: Unprocessable Entity) returned by zzzzz.arvadosapi.com:443

Without details:

    arvados API server error: 422: Unprocessable Entity returned by zzzzz.arvadosapi.com:443